staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -367,14 +376,51 @@ type QueuingConfiguration struct {
 	// +optional
 	AssuredConcurrencyShares int32 `json:"assuredConcurrencyShares" protobuf:"varint,1,opt,name=assuredConcurrencyShares"`
 
+	// `limitResponse` indicates what to do with requests that can not be executed right now
+	LimitResponse LimitResponse `json:"limitResponse,omitempty" protobuf:"bytes,2,opt,name=limitResponse"`
+}
+
+// LimitResponse defines how to handle requests that can not be executed right now.
+// +union
+type LimitResponse struct {
+	// `type` is "Queue" or "Reject".
+	// "Queue" means that requests that can not be executed upon arrival
+	// are held in a queue until they can be executed or a queuing limit
+	// is reached.
+	// "Reject" means that requests that can not be executed upon arrival
+	// are rejected.
+	// Required.
+	// +unionDiscriminator
+	Type LimitResponseType `json:"type" protobuf:"bytes,1,opt,name=type"`
```

Comment:
`Queue` can be a good default value for this field given that `Drop` is mainly for the catch-all PL

Reply:
Yes

Reply:
Arguably, "drop" is actually the current behavior, so it should be the default :)

Reply:
I think it is reasonable to declare an explicit declaration here.  If we had a default here, and a default everywhere else, that would allow a completely empty spec.  That would look odd.

Reply:
@lavalamp "drop" indeed aligns w/ the legacy max-inflight handler